### PR TITLE
(#509) Stop the WatchNotification service to hide the notification ic…

### LINKED ISSUE #509: Reply alert won't get removed until you unfollow all threads [video]
You have to stop tracking all threads to remove a notification. Also going to the thread and viewing all replies doesn't remove the notification either.

Video:
https://imgur.com/a/asCG9tb

Pixel 4 up to date

### PR DESCRIPTION
**Pull request by K1rakishou**
…on when all pins are either dead or have no unread posts

- Add a new debug setting to add a mocked reply to any post (whether it's made by you or not or it's just set as yours via the saved reply). The point is that you select "Mock reply" on any post and when a new post arrives, the post's id you marked is added to the repliesTo set of that post. You can add multiple mocked replies for the same post. In such case the next N new posts will have a reply to the post you selected, e.g.: ">>PostNoYouSelected (MOCK)". This is a really handy dev tool to debug replies to your posts.

So basically now you will be able to hide the notification by reading all posts in all active pins until the next pin update. The notification is still unswipable.

Closes #509

### CODE REVIEW

**Comment by K1rakishou** [comment]
~~Alright, I found a little bug here and I want to fix it, so don't merge this one yet.~~

File: Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/parser/CommentParser.java [hunk 1]
```diff
@@ -156,6 +171,16 @@ private CharSequence handleAnchor(
...
                 post.addReplyTo(postNo);
 
+                MockReplyManager.MockReply mockReply = mockReplyManager.getLastMockReplyOrNull(
+                        post.board.siteId,
+                        post.board.code,
+                        post.opId
+                );
+
+                if (mockReply != null) {
+                    addMockReply(theme, post, spannableStringBuilder, postNo, mockReply);
```
Comment (Adamantcheese) [inline Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/parser/CommentParser.java]: Rather than doing this whole thing here, which it looks like it would need to be attached to another quote for it to even be created, so no mocking with posts without existing quotes in them, it might be better to do this in DefaultPostParser, in parseComment, right before the TextUtils.concat call.

File: Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/parser/CommentParser.java [hunk 2]
```diff
@@ -175,13 +200,52 @@ private CharSequence handleAnchor(
...
+
+        res.setSpan(
+                pl,
+                0,
+                res.length(),
+                (250 << Spanned.SPAN_PRIORITY_SHIFT) & Spanned.SPAN_PRIORITY
+        );
+
+        post.addLinkable(pl);
+        spannableStringBuilder.append(res);
```
Comment (Adamantcheese) [inline Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/parser/CommentParser.java:256]: You can chain appends. That's kinda the point of a builder.

File: Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/parser/CommentParser.java [hunk 3]
```diff
@@ -175,13 +200,52 @@ private CharSequence handleAnchor(
...
+            Theme theme,
+            Post.Builder post,
+            SpannableStringBuilder spannableStringBuilder,
+            int postNo,
+            MockReplyManager.MockReply mockReply
+    ) {
+        Logger.d(TAG, "Adding a new mock reply (replyTo: " + mockReply.getPostNo() + ", replyFrom: " + postNo + ")");
+        post.addReplyTo(mockReply.getPostNo());
+
+        CharSequence replyText = String.format(
```
Comment (Adamantcheese) [inline Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/parser/CommentParser.java]: So uh, why even bother with a format string here? Could just be `">>" + mockReply.getPostNo() + " (MOCK)"`. Seems overcomplicated for a dev feature.

**Review by Adamantcheese — CHANGES_REQUESTED** [review]

File: Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/parser/CommentParser.java (hunk 3 above)
Comment (K1rakishou) [reply Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/parser/CommentParser.java]: What the hell. I always thought that String.format() is faster than string concatenation since it (apparently) doesn't reallocate the strings. Just checked it and turns out it isn't.

File: Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/parser/CommentParser.java (hunk 1 above)
Comment (K1rakishou) [reply Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/parser/CommentParser.java]: So I tried to do this, but it didn't work because that place is getting triggered for every post (including old posts) so the mocks are getting added to the very first posts of a thread. I need to somehow figure out that the post I want to add a new quote is in the end of a thread which I didn't even have to do when this code was in handleAnchor().

File: Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/parser/CommentParser.java (hunk 2 above)
Comment (K1rakishou) [reply Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/parser/CommentParser.java:256]: Done.

File: Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/parser/CommentParser.java (hunk 3 above)
Comment (K1rakishou) [reply Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/parser/CommentParser.java]: Done.
